lib/schema/utils.js

```diff
@@ -153,10 +156,370 @@ const parseNestedPropertyForResourceAndField = ({ arr, isValueArray }) => {
   }
 };
 
+/**
+ *
+ * @param {Object} obj
+ * @param {string} obj.string
+ * @param {string} obj.enums
+ * @param {string} obj.resourceName
+ * @param {string} obj.fieldName
+ * @param {Object} obj.schema
+ *
+ * @returns {string[]}
+ */
+const resolveStringList = ({ string, enums, schema, resourceName, fieldName }) => {
+  // if enum is of string type, no need to split
+  if (schema?.definitions?.MetadataMap?.[resourceName]?.[fieldName]?.lookupValues?.[string]?.isStringEnumeration) {
+    return [string];
+  }
+  if (enums.includes(string)) return [string];
+
+  // if type is anything other than Edm.String, we can split on commas
+  return string.split(',');
```

Comment:
In this case, it MUST be an OData IsFlags enumeration, meaning a comma-separated list of names or values. We should validate each of these items to determine whether it's an OData simple identifier, meaning it must start with a letter or underscore, followed by up to 127 letters, numbers, or underscores. Any other special characters should fail with a message that says: "Invalid OData identifier: [identifier]. See OData Simple Identifiers: https://docs.oasis-open.org/odata/odata-csdl-xml/v4.01/odata-csdl-xml-v4.01.html#_Toc38530448"